pywr-core/src/parameters/profiles/weekly.rs

```diff
@@ -0,0 +1,459 @@
+        let week_delta = current_pos - current_pos.floor();
+        first_value + (last_value - first_value) * week_delta
+    }
+
+    /// Calculate the value on the given date using the interpolation method option. In a 52-week
+    /// interpolated profile, the upper boundary in the 52nd and 53rd week is the same when
+    /// WeeklyInterpDay is First (i.e. the value on 1st January). When WeeklyInterpDay is Last the
+    /// 1st and last week will share the same lower bound (i.e. the value on the last week).
+    fn value(&self, date: &Date, interp_day: &Option<WeeklyInterpDay>) -> f64 {
+        match interp_day {
+            None => self.current(date),
+            Some(interp_day) => match interp_day {
+                WeeklyInterpDay::First => {
+                    let first_value = self.current(date);
+                    let last_value = self.next(date);
+                    self.interpolate(date, first_value, last_value)
+                }
+                WeeklyInterpDay::Last => {
+                    let first_value = self.prev(date);
+                    let last_value = self.current(date);
+                    self.interpolate(date, first_value, last_value)
+                }
+            },
+        }
+    }
+}
+
+impl From<Vec<f64>> for WeeklyProfileSize {
+    fn from(v: Vec<f64>) -> Self {
+        match v.len() {
+            52 => WeeklyProfileSize::FiftyTwo(v.try_into().unwrap()),
+            53 => WeeklyProfileSize::FiftyThree(v.try_into().unwrap()),
+            _ => panic!("52 or 53 values must be given for a weekly profile parameter"),
+        }
+    }
+}
+
+/// Schema for a weekly profile parameter. This supports a profile with either 52 or 53 weeks,
+/// with or without interpolation.
+pub struct WeeklyProfileParameter {
+    meta: ParameterMeta,
+    values: WeeklyProfileSize,
+    interp_day: Option<WeeklyInterpDay>,
+}
+
+impl WeeklyProfileParameter {
+    pub fn new(name: &str, values: Vec<f64>, interp_day: Option<WeeklyInterpDay>) -> Self {
+        Self {
+            meta: ParameterMeta::new(name),
+            values: WeeklyProfileSize::from(values),
+            interp_day,
+        }
+    }
+}
+
+impl Parameter for WeeklyProfileParameter {
+    fn as_any_mut(&mut self) -> &mut dyn Any {
+        self
+    }
+    fn meta(&self) -> &ParameterMeta {
+        &self.meta
+    }
+    fn compute(
+        &self,
+        timestep: &Timestep,
+        _scenario_index: &ScenarioIndex,
+        _model: &Model,
+        _state: &State,
+        _internal_state: &mut Option<Box<dyn Any + Send>>,
+    ) -> Result<f64, PywrError> {
+        Ok(self.values.value(&timestep.date, &self.interp_day))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::parameters::profiles::weekly::{WeeklyInterpDay, WeeklyProfileSize};
+    use float_cmp::{approx_eq, F64Margin};
+    use time::{Date, Duration, Month};
+
+    /// Compare the arrays
+    fn assert(calculated_values: &[f64], expected_values: &[f64]) {
```

Comment:
This is a helpful little function. Perhaps we can move it to the `test_utils` module so it can be reused?

Reply:
Moved to module and renamed to `assert_approx_array_eq`. The function only handles f64 but it may be make generic?